queries/highlights.scm

```diff
@@ -162,15 +168,244 @@
   "isnot"
   "cast"
 ] @keyword.operator
+; operators that are English words (e.g. `and` / `or`)
 
 [
   "return"
   "yield"
+  "discard"
 ] @keyword.return
+; keywords like `return` and `yield`
 
-; Operators
-(infix_expression operator: _ @operator)
-(prefix_expression operator: _ @operator)
 [
-  "="
-] @operator
+  "if"
+  "when"
+  "case"
+  "elif"
+  "else"
+] @conditional
+(of_branch "of" @conditional)
+; keywords related to conditionals (e.g. `if` / `else`)
+
+[
+  "for"
+  "while"
+  "continue"
+  "break"
+] @repeat
+(for "in" @repeat)
+; keywords related to loops (e.g. `for` / `while`)
+
+[
+  "block"
+] @label
+; GOTO and other labels (e.g. `label:` in C)
+
+[
+  "import"
+  "include"
+  "export"
+] @include
+(import_from_statement "from" @include)
+(except_clause "except" @include)
+; keywords for including modules (e.g. `import` / `from` in Python)
+
+[
+  "try"
+  "except"
+  "finally"
+  "raise"
+] @exception
+; keywords related to exceptions (e.g. `throw` / `catch`)
+
+((generic_parameter_list
+  (parameter_declaration
+    (symbol_declaration_list
+      (symbol_declaration name: (_) @type))))
+ (#set! "priority" 101)) ; > @parameter
+
+(call
+  (bracket_expression 
+    right: (argument_list [
+      (identifier) @type 
+      (accent_quoted (identifier) @type)
+    ]))
+)
+; match generic types when calling 
+; FIX: this also falsy matches when accessing and call elements from an array of routines
```

Comment:
This should be a note.

We can't differentiate between those, so just opt for the most probable case.

Also, please use field names if possible, it makes it easier to figure out which part are you trying to match.

Reply:
the case I'm thinking is `foo[T](bar[x])`

Reply:
in that cases the `bar[x]` is nested too deep to match anyways.

```
discard_statement [0, 0] - [0, 22]
  call [0, 8] - [0, 22]
    function: bracket_expression [0, 8] - [0, 14]
      left: identifier [0, 8] - [0, 11]
      right: argument_list [0, 12] - [0, 13]
        identifier [0, 12] - [0, 13]
    argument_list [0, 14] - [0, 22]
      bracket_expression [0, 15] - [0, 21]
        left: identifier [0, 15] - [0, 18]
        right: argument_list [0, 19] - [0, 20]
          identifier [0, 19] - [0, 20]
```